middle_end/flambda2/terms/flambda_primitive.ml

```diff
@@ -2381,9 +2229,7 @@ module Eligible_for_cse = struct
             | Make_block (Values (tag, kinds), mutability, alloc_mode) ->
               let kinds = List.map K.With_subkind.erase_subkind kinds in
               Make_block (Values (tag, kinds), mutability, alloc_mode)
-            | Make_block (Naked_floats, _, _)
-            | Make_array _ | Make_mixed_block _ ->
-              prim
+            | Make_block ((Naked_floats | Mixed _), _, _) | Make_array _ -> prim
```

Comment:
See comment on type flambda_kind.Mixed_block: if there were subkind in mixed blocks, we would need to clear them here.